crates/contracts/solidity/Trader.sol

```diff
@@ -91,7 +97,8 @@ contract Trader {
                 // revert. Instead, we fall-through so that we handle insufficient sell
                 // token balances uniformly for all tokens.
                 if (address(this).balance >= amountToWrap) {
-                    INativeERC20(nativeToken).deposit{value: amountToWrap}();
+                    try INativeERC20(nativeToken).deposit{value: amountToWrap}() {}
```

Comment:
If I understood the PR description correctly, I think this can still cause issues right? In particular, the prepare may cause the hook to revert and then you won't get accurate gas pricing on the hook execution.

Reply:
Yeah, that's true. But I'd say this would be at least a smaller issue than we currently have (many of the quotes using `permit` not getting verified).